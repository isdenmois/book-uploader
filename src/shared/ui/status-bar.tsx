import React, { useEffect } from 'react'
import { StatusBar as RNStatusBar, NativeModules, StatusBarStyle, useColorScheme } from 'react-native'

import { useTheme } from './theme'

export const StatusBar = () => {
  const mode = useColorScheme()
  const { colors } = useTheme()
  const barStyle: StatusBarStyle = mode === 'dark' ? 'light-content' : 'dark-content'

  useEffect(() => {
    NativeModules.NavigationBarColor.changeNavigationBarColor(colors.tabsBackground, mode === 'light', false)
  }, [colors, mode])

  return <RNStatusBar backgroundColor={colors.background} barStyle={barStyle} />
}
