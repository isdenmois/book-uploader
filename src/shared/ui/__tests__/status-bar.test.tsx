import React from 'react'
import { render } from '@testing-library/react-native'
import { NativeModules } from 'react-native'
import { StatusBar } from '../status-bar'

test('StatusBar', () => {
  const changeNavigationBarColor = jest.fn()
  NativeModules.NavigationBarColor = { changeNavigationBarColor }

  expect(render(<StatusBar />).toJSON()).toMatchSnapshot()
  expect(changeNavigationBarColor).toHaveBeenCalled()
})
