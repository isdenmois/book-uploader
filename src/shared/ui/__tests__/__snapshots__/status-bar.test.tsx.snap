// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar 1`] = `null`;
